sdk/containerregistry/container-registry/samples-dev/uploadDownloadBlob.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Uploads and downloads a blob to and from the repository.
+ * @azsdk-weight 3
+ */
+
+import {
+  ContainerRegistryBlobClient,
+  KnownContainerRegistryAudience,
+} from "@azure/container-registry";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+import { Readable } from "stream";
+dotenv.config();
+
+const BLOB_CONTENT = "Hello world!";
+
+async function main() {
+  // endpoint should be in the form of "https://myregistryname.azurecr.io"
+  // where "myregistryname" is the actual name of your registry
+  const endpoint = process.env.CONTAINER_REGISTRY_ENDPOINT || "<endpoint>";
+  const repository = process.env.CONTAINER_REGISTRY_REPOSITORY || "library/hello-world";
+  const client = new ContainerRegistryBlobClient(
+    endpoint,
+    repository,
+    new DefaultAzureCredential(),
+    {
+      audience: KnownContainerRegistryAudience.AzureResourceManagerPublicCloud,
+    }
+  );
+
+  // uploadBlob takes in a "resettable stream": a factory function that returns a new stream each time.
+  // It can also take in a plain ReadableStream, but this will cause the entire stream to be loaded
+  // into memory for the computation of the digest. Passing a factory function removes this limitation.
+  const uploadResult = await client.uploadBlob(() => Readable.from(BLOB_CONTENT));
+
+  // Calling downloadBlob on the uploaded blob (identified by the digest) gives a readable stream containing the blob's content.
+  const downloadResult = await client.downloadBlob(uploadResult.digest);
+
+  const downloadedBlob = await readStreamToEnd(downloadResult.content);
```

Comment:
it may be helpful to show piping the stream? I imagine common usage of  downloading to files.  Maybe in sample we can show piping to a stream that's backed by a buffer?

Reply:
Good idea. I changed it so that it pipes to the standard output, which is simpler than what I had and gets the point across IMO.